packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1023,11 +1033,19 @@ function validateTemplate(
 
     // At this point in the parsing all supported attributes from a non root template
     // should have been validated and removed from ParsedAttribute.
-    const templateAttrs = parsedAttr.getAttributes();
-    if (templateAttrs.length) {
-        ctx.warnAtLocation(ParserDiagnostics.INVALID_TEMPLATE_ATTRIBUTE, location, [
-            templateAttrs.map((attr) => attr.name).join(', '),
-        ]);
+    const invalidTemplateAttributes = parsedAttr.getAttributes();
+    if (invalidTemplateAttributes.length) {
+        // Warn when there are valid template attributes attached to the template
+        if (validTemplateAttributes) {
+            ctx.warnAtLocation(ParserDiagnostics.INVALID_TEMPLATE_ATTRIBUTE_WARNING, location, [
+                invalidTemplateAttributes.map((attr) => attr.name).join(', '),
+            ]);
```

Comment:
In the case when there are mixed valid and invalid attributes on a template, eg:

`<template if:true={showitems} class="slds-hello-world">`

Just throw a warning, as there are likely several templates that already have this mix of attributes.